ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -757,6 +767,67 @@ private AdminResponse handleCatchupStatusRequest(DataInputStream requestStream,
     return new CatchupStatusAdminResponse(isCaughtUp, adminResponse);
   }
 
+  /**
+   * Handles {@link com.github.ambry.protocol.AdminRequestOrResponseType#StopBlobStore}.
+   * @param requestStream the serialized bytes of the request.
+   * @param adminRequest the {@link AdminRequest} received.
+   * @return the {@link AdminResponse} to the request.
+   * @throws IOException if there is any I/O error reading from the {@code requestStream}.
+   */
+  private AdminResponse handleStopBlobStoreRequest(DataInputStream requestStream, AdminRequest adminRequest)
+      throws IOException {
+    ServerErrorCode error;
+    StopBlobStoreAdminRequest stopBlobStoreAdminRequest =
+        StopBlobStoreAdminRequest.readFrom(requestStream, adminRequest);
+    if (stopBlobStoreAdminRequest.getNumReplicasCaughtUpPerPartition() > 0) {
+      PartitionId partitionId = stopBlobStoreAdminRequest.getPartitionId();
+      if (partitionId != null) {
+        error = validateRequest(partitionId, RequestOrResponseType.AdminRequest);
+        if (error.equals(ServerErrorCode.No_Error)) {
+          Collection<PartitionId> partitionIds = Collections.singletonList(partitionId);
+          if (storageManager.disableCompactionForBlobStore(partitionId)) {
+            controlRequestForPartitions(RequestOrResponseType.PutRequest, partitionIds, false);
+            controlRequestForPartitions(RequestOrResponseType.DeleteRequest, partitionIds, false);
+            if (replicationManager.controlReplicationForPartitions(partitionIds, Collections.<String>emptyList(),
+                false)) {
+              if (isRemoteLagLesserOrEqual(partitionIds, 0,
+                  stopBlobStoreAdminRequest.getNumReplicasCaughtUpPerPartition())) {
+                controlRequestForPartitions(RequestOrResponseType.GetRequest, partitionIds, false);
+                controlRequestForPartitions(RequestOrResponseType.ReplicaMetadataRequest, partitionIds, false);
+                // Shutdown the BlobStore completely
+                if (storageManager.shutdownBlobStore(partitionId)) {
+                  error = ServerErrorCode.No_Error;
```

Comment:
let's add a trace message here saying "store shutdown for partition: {}"

Reply:
trace message added.